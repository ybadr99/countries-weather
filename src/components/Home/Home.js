import React from 'react';
import './Home.scss';
import Card from '../Card/Card';
import countries from '../../data.json';

const Home = () => (
  <div className="body">
    {countries.map((country) => (
      <Card key={country.name} country={country} />
    ))}
  </div>
);

export default Home;
