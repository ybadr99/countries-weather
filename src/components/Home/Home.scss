@import "../../style/variables.scss";

.home {
  width: 100%;
  background-color: $very-dark-blue;
  padding: 1.5rem 4rem;
}

.cards {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  margin-top: 2rem;
  gap: 2rem 2rem;
}

nav {
  display: flex;
  justify-content: space-between;
}

.dark-btn {
  font-size: 1.5rem;
  cursor: pointer;
}
