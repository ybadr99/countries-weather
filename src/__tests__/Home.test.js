import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { MemoryRouter } from 'react-router-dom';
import Home from '../components/Home/Home';
import countriesReducer from '../redux/countries/countriesSlice';
import '@testing-library/jest-dom/';

it('renders the countries correctly', () => {
  const countries = [
    {
      name: 'Afghanistan',
      topLevelDomain: ['.af'],
    },
    {
      name: 'Ã…land Islands',
      topLevelDomain: ['.ax'],
    },
  ];

  const rootReducer = combineReducers({
    countries: countriesReducer,
  });

  const store = configureStore({
    reducer: rootReducer,
    preloadedState: {
      countries: {
        countries,
        status: 'success',
        error: null,
      },
    },
  });

  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
