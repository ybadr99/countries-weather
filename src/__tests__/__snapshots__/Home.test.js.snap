// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the countries correctly 1`] = `
<div>
  <nav>
    <h1>
      Where in the world?
    </h1>
  </nav>
  <button
    class="filter"
    type="button"
  >
    <div>
      Filter by region
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
        />
      </svg>
    </div>
  </button>
  <div
    class="body"
  >
    <div
      aria-hidden="true"
      class="card"
    >
      <img
        alt="Flag"
      />
      <ol>
        <h2>
          Afghanistan
        </h2>
        <li>
          Population:
          <span>
            NaN
          </span>
        </li>
        <li>
          Region:
          <span />
        </li>
        <li>
          Capital:
          <span />
        </li>
      </ol>
    </div>
    <div
      aria-hidden="true"
      class="card"
    >
      <img
        alt="Flag"
      />
      <ol>
        <h2>
          Ã…land Islands
        </h2>
        <li>
          Population:
          <span>
            NaN
          </span>
        </li>
        <li>
          Region:
          <span />
        </li>
        <li>
          Capital:
          <span />
        </li>
      </ol>
    </div>
  </div>
</div>
`;
